src/main/java/org/seqdoop/hadoop_bam/BAMInputFormat.java

```diff
@@ -308,60 +303,79 @@ private int addProbabilisticSplits(
 					.setUseAsyncIo(false);
 			SamReader samReader = readerFactory.open(NIOFileUtil.asPath(fs.makeQualified(bamFile).toUri()));
 			if (!samReader.hasIndex()) {
-				System.err.println("WARNING: no BAM index file found, splits will not be " +
-						"filtered, which may be very inefficient: " + bamFile);
-				return splits;
+				throw new IllegalArgumentException("Intervals set but no BAM index file found for " + bamFile);
```

Comment:
I would suggest to lift the reader factory construction out of the loop, and acquire the reader in a try-with resources.